src/filesystem.cpp

```diff
@@ -685,194 +670,83 @@ static std::optional<std::string> get_games_path()
 		path = games_path.string();
 	} else {
 		ERR_FS << "Could not determine path to user's Documents folder! (" << std::hex << "0x" << res << std::dec << ") "
-				<< "User config/data directories may be unavailable for "
-				<< "this session. Please report this as a bug.";
+				<< "Please report this as a bug.";
 	}
 
 	CoTaskMemFree(docs_path);
 	return path;
 }
+#endif
 
-static bfs::path windows_userdata(const std::string& newprefdir)
+void set_user_data_dir(std::string newprefdir)
 {
-	bfs::path dir;
-	std::string temp = newprefdir;
-
 #ifdef PREFERENCES_DIR
-	if(temp.empty()) {
-		temp = PREFERENCES_DIR;
+	if(newprefdir.empty()) {
+		newprefdir = PREFERENCES_DIR;
 		DBG_FS << "Using PREFERENCES_DIR '" << PREFERENCES_DIR << "'";
 	}
 #endif
 
-	// if a custom userdata directory is provided as an absolute path, just use that
-	// else if it's relative to the current working directory, just use that
-	// else if the first characters is a tilde, then replace that with the path to the "My Games" folder
-	// else if no custom userdata directory was provided, default to the "My Games" folder if present or fallback to the current working directory if not
-	// else a relative path was provided that isn't relative to the current working directory, so ignore it and use the default
```

Comment:
Although the logic has been simplified, it's still complex enough that an updated version of this summary should be documented here.

Reply:
`// if no custom userdata directory was provided, use appropriate default`
Is that not enough for the first if? IMO it's pretty easy to see the defaults for the different cases.

The second if that handles tilde expansion could be documented like:
`// replace ~ with Documents/My Games on windows and $HOME otherwise`

Reply:
I'd like to have `// we'll replace ~ with Documents/My Games on Windows and $HOME otherwise` here. It makes the first half of the function easier to read, instead of having a "eh? Tilde expansion on Windows?" moment.

Reply:
You mean because the default path on windows contains the tilde? Sure, I can do that.